src/databricks/labs/lsql/dashboards.py

```diff
@@ -893,6 +896,46 @@ def _from_dashboard_folder(cls, folder: Path) -> "DashboardMetadata":
             tiles.append(tile)
         return cls(display_name=folder.name, _tiles=tiles)
 
+    @staticmethod
+    def _add_to_zip(file_path: Path) -> None:
+        """Add a file to a ZIP archive, handling duplicates by renaming."""
+        zip_path = file_path.parent / "export_to_zipped_csv.zip"
+
+        try:
+            with ZipFile(zip_path, "a") as zip_file:
+                zip_file.write(file_path, arcname=file_path)
+
+        except FileNotFoundError:
+            logger.warning(f"File {file_path} not found.")
+        except PermissionError:
+            logger.warning(f"Permission denied for {file_path} or {zip_path}.")
+
+        # Clean up the file if it was successfully added
+        if file_path.exists():
+            file_path.unlink()
+
+    def export_to_zipped_csv(
+        self, sql_backend: SqlBackend, queries_path: Path, export_path: Path, catalog: str, database: str
+    ) -> Path:
+        """Export the dashboard queries to CSV files in a ZIP archive."""
+        dashboard = self.from_path(queries_path)
+        dashboard = dashboard.replace_database(catalog=catalog, database=database)
+        for tile in dashboard.tiles:
```

Comment:
```suggestion
    def export_to_zipped_csv(self, sql_backend: SqlBackend, export_path: Path) -> Path:
        """Export the dashboard queries to CSV files in a ZIP archive."""
        for tile in self.tiles:
```

why do you need to call `from_path` if you already have a dashboard metadata instance?

Reply:
Function is using the Dashboard Metadata instance to avoid additional parameters.